main.cpp

```diff
@@ -19,13 +19,34 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter &rewriter) : rewriter_(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *castExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        const auto castIgnore = castExpr->getSubExprAsWritten()->IgnoreImpCasts();

```

Comment:
Please describe why you are using here `getSubExprAsWritten()` and `IgnoreImpCasts()`.
What would be the difference with the use of `getSubExprAsWritten()->IgnoreImpCasts()`, `getSubExpr()->IgnoreImpCasts()` and `getSubExprAsWritten()`?

Reply:
1. `getSubExprAsWritten()` retrieves the cast subexpression as it was written in the source code, looking through any implicit casts or other intermediate nodes.
2. `getSubExpr()` retrieves the first cast subexpression.
3. `ignoreImpCasts()` skips any implicit casts which might surround expression

Since the current task doesn’t contain implicit type casts, it isn’t necessary to use `getSubExprAsWritten() -> IgnoreImpCasts()`. This part of the code can be written as follows: `getSubExprAsWritten()` or `getSubExpr() -> IgnoreImpCasts()`. 

When `getSubExpr()` used without `IgnoreImpCasts()`, we get not what is expected.

Reply:
`Since the current task doesn’t contain implicit type casts` - this suggestion is incorrect. You can dump AST to be sure.

Reply:
Yes, you're right.
```
llvm::outs() << сastExpr -> getSubExprAsWritten() -> IgnoreImpCasts() -> getStmtClassName();
// Result is DeclRefExpr
```
```
llvm::outs() << сastExpr -> getSubExpr() -> IgnoreImpCasts() -> getStmtClassName();
// Result is DeclRefExpr
```
```
llvm::outs() << сastExpr -> getSubExprAsWritten() -> getStmtClassName();
// Result is DeclRefExpr
```
The presence of an implicit cast is seen in the following output:
```
llvm::outs() << сastExpr -> getSubExpr -> getStmtClassName();
// Result is ImplicitCastExpr
```